packages/compartment-mapper/src/import-hook.js

```diff
@@ -1,11 +1,27 @@
 // @ts-check
+
+/** @typedef {import('ses').ImportHook} ImportHook */
+/** @typedef {import('ses').StaticModuleType} StaticModuleType */
+/** @typedef {import('./types.js').ReadFn} ReadFn */
+/** @typedef {import('./types.js').Sources} Sources */
+/** @typedef {import('./types.js').CompartmentDescriptor} CompartmentDescriptor */
+/** @typedef {import('./types.js').ImportHookMaker} ImportHookMaker */
+
 import { parseExtension } from './extension.js';
 
 // q, as in quote, for quoting strings in error messages.
 const q = JSON.stringify;
 
 const { apply } = Reflect;
-const { freeze } = Object;
+
+/**
+ * TypeScript cannot be relied upon to deal with the nuances of Readonly, so we
+ * borrow the pass-through type definition of harden here.
```

Comment:
Isn't it dangerous if we ever update the definition of `Harden` to be more correct? I might be of the opinion to just be explicit here by copying the current passthrough definition of `Harden`

Reply:
I figured out that it will be a devastating breaking change to make the `Harden` type definition correct regardless. It’s the kind of breaking change that would require TypeScript to grow a new head before we could even contemplate it. I don’t think this digs our heels in too badly. If we do introduce a `ProperHarden` type, it’ll need a different name and a long migration period to propagate outward.